libvips/iofuncs/image.c

```diff
@@ -1658,18 +1658,22 @@ vips_image_set_kill( VipsImage *image, gboolean kill )
 	image->kill = kill;
 }
 
-/* Make a name for a filename-less image. Use immediately, don't free the
- * result.
+/* Fills the given buffer with a temporary filename.
+ * Assuming that "int" might be 64 Bit wide a buffer size of 26 suffices.
  */
-static const char *
-vips_image_temp_name( void )
+void
+vips_image_temp_name( char *name, int size )
 {
-	static int serial = 0;
-	static char name[256];
+	static int global_serial = 0;
 
-	vips_snprintf( name, 256, "temp-%d", serial++ );
+	int serial =
+#if GLIB_CHECK_VERSION( 2, 30, 0 )
+		g_atomic_int_add( &global_serial, 1);
+#else
+		g_atomic_exchange_and_add( &global_serial, 1);
```

Comment:
`g_atomic_exchange_and_add` -> [`g_atomic_int_exchange_and_add`](https://developer.gnome.org/glib/stable/glib-Atomic-Operations.html#g-atomic-int-exchange-and-add)

Reply:
Oops! I'll fix it. Nice spot Kleis.

Reply:
Oh dear... I'm sorry for this mistake. :disappointed: